### PR TITLE
drivers: sensor: adi: add the ADE9153a AC energy metering IC driver

### PR DESCRIPTION
**Pull request by rodrigopex**
Add the ADE9153a AC energy metering IC driver to Zephyr.

Work to be finished: 

- [x] Fix compliance issues
- [ ] Driver and sample documentation
- [ ] Complete kconfig help fields missing
- [ ] Add to the sample a way of activating the IRQ or CF trigger

### CODE REVIEW

File: drivers/sensor/adi/ade9153a/ade9153a.c [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+		break;
+	}
+	case SENSOR_CHAN_AC_POWER_FACTOR: {
+		/* Calculate PF */
+		double_value = (double)data->power_factor_reg / (double)134217728;
+		break;
+	}
+	case SENSOR_CHAN_AC_PERIOD: {
+		/* Calculate Frequency */
+		double_value = (double)(4000 * 65536) / (double)(data->period_reg + 1);
```
Comment (MikePlentify) [inline drivers/sensor/adi/ade9153a/ade9153a.c:245]: @rodrigopex In the [Datasheet Page 27](https://www.analog.com/media/en/technical-documentation/data-sheets/ade9153a.pdf), it gives the `Line Period Calculation` as the inverse of what you have here. 

It looks like both your `SENSOR_CHAN_AC_PERIOD`  & `SENSOR_CHAN_AC_FREQUENCY` give the same value with Frequency just scaled by `100`.

Should it not look like:
```
case SENSOR_CHAN_AC_PERIOD: {
	/* Calculate Line Period */
	double_value = (double)(data->period_reg + 1) / (double)(4000 * 65536);
	break;
}
case SENSOR_CHAN_AC_FREQUENCY: {
	/* Calculate Frequency in cHz*/
	double_value = (4000LU * 65536 * 100) / (double)(data->period_reg + 1);
	break;
}
```

File: include/zephyr/drivers/sensor/ade9153a.h [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+/* Power update time configuration. */
+#define ADE9153A_REG_PWR_TIME         0x04B1
+/* Energy accumulation update time configuration. */
+#define ADE9153A_REG_EGY_TIME         0x04B2
+/* This register forces an update of the CRC of configuration registers. */
+#define ADE9153A_REG_CRC_FORCE        0x04B4
+/* Temperature sensor configuration register. */
+#define ADE9153A_REG_TEMP_CFG         0x04B6
+/* Temperature measurement result. */
+#define ADE9153A_REG_TEMP_RSLT        0x04B7U
```
Comment (MikePlentify) [inline include/zephyr/drivers/sensor/ade9153a.h:277]: @rodrigopex This value is defined with the suffix `U` - differently to the rest of the register addresses in this header file. Is this intentional?

Comment (rodrigopex) [reply include/zephyr/drivers/sensor/ade9153a.h:277]: I will fix that. Thanks.

File: drivers/sensor/adi/ade9153a/ade9153a.c (hunk 1 above)
Comment (rodrigopex) [reply drivers/sensor/adi/ade9153a/ade9153a.c:245]: Ok. I will fix that.

File: drivers/sensor/adi/ade9153a/ade9153a.c [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+
+static inline int ade9153a_set_reg32(const struct device *dev, uint16_t reg_addr, uint32_t data)
+{
+	uint8_t data_raw[4] = {(uint8_t)data >> 24U, (uint8_t)data >> 16U, (uint8_t)data >> 8U,
+			       (uint8_t)data};
+
+	return ade9153a_reg_access(dev, ADE9153A_WRITE_REG, reg_addr, data_raw, 4U);
+}
+
+int start_acal(const struct device *dev, int32_t reg_ms_acal_cfg)
```
Comment (MikePlentify) [inline drivers/sensor/adi/ade9153a/ade9153a.c:106]: @rodrigopex The argument `reg_ms_acal_cfg` is set as a signed integer but represents an unsigned value being written to a register to configure the type of Msure Acal to run - is this intentional?

File: drivers/sensor/adi/ade9153a/ade9153a.c (hunk 1 above)
Comment (rodrigopex) [reply drivers/sensor/adi/ade9153a/ade9153a.c:245]: Done.

File: drivers/sensor/adi/ade9153a/ade9153a.c (hunk 3 above)
Comment (rodrigopex) [reply drivers/sensor/adi/ade9153a/ade9153a.c:106]: Changed to `uint32_t`.

File: drivers/sensor/adi/ade9153a/ade9153a_trigger.c [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+int ade9153a_trigger_set(const struct device *dev, const struct sensor_trigger *trig,
+			 sensor_trigger_handler_t handler)
+{
+	__ASSERT_NO_MSG(trig->chan == SENSOR_CHAN_ALL);
+
+	struct ade9153a_data *data = dev->data;
+	const struct ade9153a_config *config = dev->config;
+	enum sensor_trigger_ade9153a ade_trig_type = (enum sensor_trigger_ade9153a)trig->type;
+
+	if (!config->irq_gpio_dt_spec.port && !config->cf_gpio_dt_spec.port) {
```
[the remark this answers is not in the document]
Comment (rodrigopex) [reply drivers/sensor/adi/ade9153a/ade9153a_trigger.c:84]: Please take a look and check if it is okay.

**Comment by ttmut** [comment]
You should rebase instead of merging, otherwise compliance check will fail.

**Comment by rodrigopex** [comment]
@ttmut 
> You should rebase instead of merging, otherwise compliance check will fail.

I was by accident. I was a GitHub thing. I pressed the wrong button there. Thanks for pointing that out.

**Pull request closed without merging**